backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/DiagramEventFlux.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.spring.collaborative.diagrams;
+
+import java.util.Objects;
+
+import org.eclipse.sirius.web.collaborative.api.dto.PreDestroyPayload;
+import org.eclipse.sirius.web.collaborative.diagrams.api.dto.DiagramRefreshedEventPayload;
+import org.eclipse.sirius.web.diagrams.Diagram;
+import org.eclipse.sirius.web.services.api.dto.IPayload;
+
+import reactor.core.publisher.DirectProcessor;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+import reactor.core.publisher.Mono;
+
+/**
+ * Service used to manage the diagram event flux.
+ *
+ * @author sbegaudeau
+ */
+public class DiagramEventFlux {
```

Comment:
I was torn on the issue since this was part of the representation event processor at the beginning and it may not make a lot of sense to keep it separated since it's so small. We could also indeed make it generic and update other representations along with the frontend to have a generic RepresentationRefreshedEventPayload.

If it's ok with you, we could go with such a solution in a dedicated issue since it will involve modifying quite a lot of code (including frontend reducers etc) in a commit about unsynchronized diagrams.